### PR TITLE
:seedling: apibinding, types: remove ancestor check

### LINKED ISSUE #2059: bug: Cross-workspace binding without ancestor does not work
### Describe the bug

If you try create binding from `:corp:compute:services` to `:corp:services:warehouse` it fails with:
`error starting server: apibindings.apis.kcp.dev "kubernetes" is forbidden: spec.reference.workspace.path: not pointing to an ancestor or child of an ancestor of "root:corp:services:warehouse"exit status 1`

due to this https://github.com/kcp-dev/kcp/blob/main/pkg/admission/apibinding/apibinding_admission.go#L180 

```
kubectl kcp workspace tree
.
└── root
    ├── corp
    │   ├── compute
    │   │   ├── services
    │   │   └── shared
    │   └── services
    │       └── warehouse
    └── users
```

As per https://kubernetes.slack.com/archives/C021U8WSAFK/p1663861982022829 it should be removed.

### Steps To Reproduce

See thread in slack

### Expected Behaviour

Working. 

### Additional Context

cc @s-urbaniak

### PR DESCRIPTION
**Pull request by s-urbaniak**
## Summary
This removes ancestor checks for API bindings.

## Related issue(s)

Follow-up for #1739
Fixes #2059

### CODE REVIEW

File: pkg/admission/apibinding/apibinding_admission.go [hunk 1]
```diff
@@ -165,22 +165,7 @@ func (o *apiBindingAdmission) Validate(ctx context.Context, a admission.Attribut
...
-	case apiBinding.Spec.Reference.Workspace.Path != "":
-		absoluteRef := logicalcluster.New(apiBinding.Spec.Reference.Workspace.Path)
-		absoluteRefParent, _ := absoluteRef.Parent()
-		isAncestor := cluster.Name.HasPrefix(absoluteRef)
-		isAncestorChild := cluster.Name != absoluteRefParent && cluster.Name.HasPrefix(absoluteRefParent)
-		if !isAncestor && !isAncestorChild {
-			return admission.NewForbidden(a, fmt.Errorf("spec.reference.workspace.path: not pointing to an ancestor or child of an ancestor of %q", cluster.Name))
-		}
-		apiExportClusterName = absoluteRef
-	default:
```
Comment (sttts) [inline pkg/admission/apibinding/apibinding_admission.go:183]: wondering, the docs say that the path is defaulted. Where does that happen?

File: config/crds/workload.kcp.dev_synctargets.yaml [hunk 2]
```diff
@@ -102,9 +102,8 @@ spec:
                           type: string
                         path:
                           description: path is an absolute reference to a workspace,
-                            e.g. root:org:ws. The workspace must be some ancestor
-                            or a child of some ancestor. If it is unset, the path
-                            of the APIBinding is used.
+                            e.g. root:org:ws. If it is unset, the path of the APIBinding
+                            is used.
```
Comment (sttts) [inline config/crds/workload.kcp.dev_synctargets.yaml:106]: there was no code checking this?

**Comment by sttts** [comment]
Missing the changes for ClusterWorkspaceTypes.

File: config/crds/workload.kcp.dev_synctargets.yaml (hunk 2 above)
Comment (s-urbaniak) [reply config/crds/workload.kcp.dev_synctargets.yaml:106]: there was, this is the expected state.

**Comment by s-urbaniak** [comment]
> Missing the changes for ClusterWorkspaceTypes.

it seems there is nothing to be done here.

**Review by stevekuznetsov — APPROVED** [review]

File: pkg/admission/apibinding/apibinding_admission.go (hunk 1 above)
Comment (ncdc) [reply pkg/admission/apibinding/apibinding_admission.go:183]: It's in apibinding_admission.go

**Pull request merged**